bin/endorS.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python3
+
+"""Script to calculate the endogenous DNA in a sample from samtools flag stats.
+It accepts can accept up to two files: pre-quality and post-quality filtering. We recommend
+to use both files but you can also use the pre-quality filtering.
+"""
+import re
+import sys
+import json
+import argparse
+import textwrap
+
+parser = argparse.ArgumentParser(prog='endorS.py', 
+   usage='python %(prog)s [-h] [--version] <samplesfile>.stats [<samplesfile>.stats]',
+   formatter_class=argparse.RawDescriptionHelpFormatter,
+   description=textwrap.dedent('''\
+   author: 
+     Aida Andrades Valtueña (aida.andrades[at]gmail.com)
+   
+   description:
+     %(prog)s calculates endogenous DNA from samtools flagstat files and print to screen
+     Use --output flag to write results to a file
+   '''))   
+parser.add_argument('samtoolsfiles', metavar='<samplefile>.stats', type=str, nargs='+',
+                    help='output of samtools flagstat in a txt file (at least one required). If two files are supplied, the mapped reads of the second file is divided by the total reads in the first, since it assumes that the <samplefile.stats> are related to the same sample. Useful after BAM filtering')
+parser.add_argument('-v','--version', action='version', version='%(prog)s 0.1')
+parser.add_argument('--output', '-o', nargs='?', help='specify a file format for an output file. Options: <json> for a MultiQC json output. Default: none')
+parser.add_argument('--name', '-n', nargs='?', help='specify name for the output file. Default: extracted from the first samtools flagstat file provided')
+args = parser.parse_args()
+
+#Open the samtools flag stats pre-quality filtering:
+try:
+    with open(sys.argv[1], 'r') as pre:
```

Comment:
Can you parse all the input files and arguments with argparse ?
Something like this:
Just switch to `parser.add_argument('samtools_pre', help="flag stats pre-quality filtering")`
Here is an [example](https://github.com/maxibor/adrsm/blob/826bcd7ec9ed271401cc5cc4c0d0a545050eb65a/adrsm#L8) with argparse.

Reply:
@aidaanva